src/containers/edit-profile/profile-tab/ProfileTab.tsx

```diff
@@ -1,16 +1,123 @@
-import { FC } from 'react'
-import { ProfileProvider } from '~/context/profile-context'
-import { ProfileTabProps } from '~/types'
-import ProfileTabInnerContainer from '~/containers/edit-profile/profile-tab/profile-tab-inner-container/ProfileTabInnerContainer'
-import { getProfileContextInitialValues } from '~/utils/get-profile-context-initial-values'
+import { FC, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Box } from '@mui/material'
+import { useBlocker } from 'react-router-dom'
+
+import useUpdateUser from '~/hooks/use-update-user'
+import useConfirm from '~/hooks/use-confirm'
+import useForm from '~/hooks/use-form'
+
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import AppButton from '~/components/app-button/AppButton'
+import ProfileTabForm from '~/containers/edit-profile/profile-tab/profile-tab-form/ProfileTabForm'
+import {
+  getProfileInitialValues,
+  getUserUpdatedData
+} from '~/utils/get-profile-values'
+import { validations } from '~/components/user-steps-wrapper/constants'
+import { styles } from '~/containers/edit-profile/profile-tab/ProfileTab.styles'
+import {
+  ButtonVariantEnum,
+  EditProfileForm,
+  SizeEnum,
+  UpdatedPhoto,
+  UserResponse
+} from '~/types'
+
+interface ProfileTabProps {
+  user: UserResponse
+}
 
 const ProfileTab: FC<ProfileTabProps> = ({ user }) => {
-  const initialGeneralData = getProfileContextInitialValues(user)
+  const { t } = useTranslation()
+  const { setNeedConfirmation, checkConfirmation } = useConfirm()
+  const { handleSubmit, loading } = useUpdateUser(user._id)
+  const [photo, setPhoto] = useState<UpdatedPhoto | null>({
+    src: '',
+    name: ''
+  })
+
+  const initialValues = getProfileInitialValues(user)
+
+  const {
+    isDirty,
+    handleInputChange,
+    handleBlur,
+    handleNonInputValueChange,
+    data,
+    errors
+  } = useForm<EditProfileForm>({
+    initialValues,
+    validations
+  })
+
+  const isPhotoChanged = photo === null || Boolean(photo.src)
+  const isDataChanged = isDirty || isPhotoChanged
+
+  const blocker = useBlocker(isDataChanged)

```

Comment:
Very interesting solution! I have never heard about this hook before

Reply:
I used it for the first time as well, and it turned out to be quite useful in cases where we need to block navigation from a particular page for some reason.